src/gConsent/manage/revokeAccessGrant.ts

```diff
@@ -38,29 +39,27 @@ import { getBaseAccess } from "../util/getBaseAccessVerifiableCredential";
 const { quad, namedNode } = DataFactory;
 
 /**
- * Makes a request to the access server to revoke a given Verifiable Credential (VC).
- *
- * @param vc Either a VC, or a URL to a VC, to be revoked.
- * @param options Optional properties to customise the request behaviour.
- * @returns A void promise.
- * @since 0.4.0
+ * @internal
  */
-async function revokeAccessGrant(
+async function revokeAccessCredential(
   vc: DatasetWithId | VerifiableCredential | URL | UrlString,
   options: Omit<AccessBaseOptions, "accessEndpoint"> = {},
+  type: NamedNode<string>[] = [
+    solidVc.SolidAccessGrant,
+    solidVc.SolidAccessDenial,
+  ],
 ): Promise<void> {
-  const credential = await getBaseAccess(vc, options, solidVc.SolidAccessGrant);
+  const credential = await getBaseAccess(vc, options, type[0]);
 
   if (
-    !credential.has(
-      quad(namedNode(getId(credential)), TYPE, solidVc.SolidAccessGrant),
-    ) &&
-    !credential.has(
-      quad(namedNode(getId(credential)), TYPE, solidVc.SolidAccessDenial),
+    type.every(
+      (vcType) =>
+        !credential.has(quad(namedNode(getId(credential)), TYPE, vcType)),
     )
   ) {
     throw new Error(
-      `An error occurred when type checking the VC: Not of type [${solidVc.SolidAccessGrant.value}] or [${solidVc.SolidAccessDenial.value}].`,
+      `An error occurred when type checking the VC: Not of type [${type[0].value}] 
+      ${type.length > 1 ? `or [${type[1].value}]` : ``}.`,
     );
   }
```

Comment:
`getBaseAccess` already checks for the type, so it is actually not be necessary at all to perform the check again (which was already present before your PR, it's nothing you've introduced no worries).

Provided the underlying `getBaseAccess` checks for the type and throws on mismatch, I would be tempted to do something like: 
```
  const credential = (
    await Promise.all(
      types.map(async (type) => {
        try {
          return await getBaseAccess(vc, options, type);
        } catch (e) {
          return undefined;
        }
      }),
    )
  ).find((cred) => cred !== undefined);
```
and then test if credential is undefined. If so, the VC was not of the target types, and you can throw an error as previously, and otherwise you can carry on with the rest of the function. Does this make sense?

Reply:
Indeed, although you'll want to do something more generic rather than expecting one or two types: 
```
`An error occurred when type checking the VC: Not of the expected types ${types.map((type) => type.value)}`
```
and then line 62 becomes unnecessary.

Reply:
There's an "issue" with the new code for computing the credential: there are some specific tests that are expecting the Error thrown by `getBaseAccess`. Because of the `try/catch` this error is handled, and hence the message from the error thrown is not the expected `401 Unauthorized` but the one introduced in line 61.

Should I go ahead and adapt the tests or should we go back to the previous version?